app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".email_reset_password"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t ".email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t(".password_empty")
+      render :edit
+    elsif @user.update(user_params)
+      log_in @user
+      @user.update_attribute(:reset_digest, nil)
+      flash[:success] = t ".password_has_been_reset"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def get_user
+    @user = User.find_by email: params[:email]
+  end
+
+  def valid_user
+    unless @user&.activated? &&
+           @user.authenticated?(:reset, params[:id])
+      redirect_to root_url
```

Comment:
thêm message nói tại sao lại bị chuyển sang trang root chứ em, cho nó clear ng dùng

Toàn bộ trong chương trình thì nên dùng `_path` thay vì `_url`, trừ trường hợp call  bên ngoài proejct như gửi mail